docs/developer-guide/etendo-classic/how-to-guides/how-to-create-junit-testcases.md

```diff
@@ -0,0 +1,525 @@
+final List<User> uList = uCriteria.list();
+```
+
+Uses the  OBDal  instance to create a new  OBCriteria  object, and uses it for
+listing all (since we are not filtering) the Users in the database.
+
+```java
+int userCount = 0;
+for (User u: uList) {
+  if (u != null && u.getPassword() != null && !u.getPassword().isEmpty())
+    userCount++;
+}
+```
+
+We loop trough the uList collection, and we increment the userCount variable if the user has a password.
+
+```java
+assertTrue(userCount > 0);
+```
+
+We assert that the userCount is more than 0.
+
+```java
+System.out.println("Total of users with password: " + (userCount));
+```
+
+Finally we print a total of users with password just for the record.
+
+###  Transaction Handling
+
+A question which might pop-up when looking at the above code: where is the
+database transaction handling done? The answer is that this is handled by the
+WeldBaseTest class and the Openbravo data access layer:
+
+  * a transaction is automatically started at first database access in the test cases. This is done by the Data Access Layer. 
+  * a transaction is either committed (when no exception happened) or rolled-back (when an exception happened). 
+
+The WeldBaseTest class detects automatically if an exception happened or not.
+
+There are certainly cases whereby it makes sense to have more control over the
+database transactions. There are a number of relevant methods which can be
+useful then:
+
+  * **OBDal.getInstance().flush()**: flushes the update/insert queries in hibernate to the database. 
+  * **OBDal.getInstance().commitAndClose()**: commits the transaction and closes the session. A new session/transaction is automatically started at the next database access. 
+  * **OBDal.getInstance().rollbackAndClose()**: rolls back and closes the transactions. A new session/transaction is automatically started at the next database access. 
+
+!!! info
+    DAL event observers  are not triggered within test cases extending ` OBBaseTest ` class. They to work require of test cases extending ` WeldBaseTest `. 
+
+###  Side-Effect Free
+
+A test case will often change the data in the underlying database. Most of the
+time it is not feasible to setup a completely new test database for each test
+run. Therefore test-cases should be developed such that they are side effect
+free. This means:
+
+  * When the test-case changes data then it should have a test method which is run as the last test method which cleans up/repairs the data. 
+  * This clean-up method should also clean up data which is left from previous test runs. For this common issue should be used _AfterClass_ notation. This method runs automatically at the end of the class.
+
+This last point is important because there can be always reasons why during a
+test the clean-up step is not performed. For example because the test run is
+stopped before the clean-up is done.
+
+###  Approach to features of JUnit 4.11
```

Comment:
Has the current Junit library been upgraded to Junit5?

Reply:
some things in JUnit 5 don't work well, I can delete the version if you like